### PR TITLE
Issue/68 create site 

### LINKED ISSUE #68: create a `user`, `site` and `site group`
## Detailed Description
Add three functions to 
- create user
- create site
- create site group

## Context
- easy way to make users, sites and site group

## possible implentation
- show example json data
- let user change the text
- press submit

### PR DESCRIPTION
**Pull request by rachel-labri-tipton**
# Pull Request

## Description

Add a feature to create a new site in the database. I've added an expander which opens and then data added using text input. 

Screenshot 2023-09-25 at 16 52 16

Screenshot 2023-09-25 at 16 52 49

Fixes #68 

## How Has This Been Tested?

Ran the code locally and wrote a passing pytest. 

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/sites_toolbox.py [hunk 1]
```diff
@@ -297,3 +300,30 @@ def sites_toolbox_page():
...
+        f'<h1 style="color:#63BCAF;font-size:32px;">{"Create New Site"}</h1>',
+        unsafe_allow_html=True,
+    )
+    if st.button("Input new site data"):
+        with connection.get_session() as session:
+            client_site_id = st.text_input("Enter client_site_id")
+            client_site_name = st.text_input("Enter client_site_name")
+            latitude = st.text_input("Enter latitude")
+            longitude = st.text_input("Enter longitude")
+            capacity_kw = st.text_input("Enter capacity_kw")
```
Comment (peterdudfield) [inline src/sites_toolbox.py:315]: Could you an some optional variables, like tilt and orientation? Perhaps there should be a red start with the ones that are needed

File: src/get_data.py [hunk 2]
```diff
@@ -180,3 +185,42 @@ def add_site_to_site_group(
     session.commit()
 
     return site_group.sites
+
+
+# make site
+def create_new_site(
+    session: Session,
+    ml_id=int,
```
Comment (peterdudfield) [inline src/get_data.py:193]: I think it should be `ml_id:int` not `ml_id=int` and for others

File: tests/test_get_data.py [hunk 3]
```diff
@@ -42,3 +47,25 @@ def test_add_site_to_site_group(db_session):
...
+        created_utc="2021-01-01 00:00:00",
+    )
+
+    assert site.client_site_name == "test_site_name"
+    assert site.ml_id == 1
+    assert site.client_site_id == 6932
+    assert (
+        message
+        == f"Site with client site id {site.client_site_id} and site uuid {site.site_uuid} created successfully"
+    )
```
Comment (peterdudfield) [inline tests/test_get_data.py:71]: Can you add a test that adds two sites, to check `ml_id` thing works

**Review by peterdudfield — CHANGES_REQUESTED** [review]
few small changes, but looks good

File: src/sites_toolbox.py [hunk 4]
```diff
@@ -297,3 +300,30 @@ def sites_toolbox_page():
...
+            created_utc = datetime.now(timezone.utc).strftime("%Y-%m-%d %H:%M:%S")
+            if st.button("Create site"):
+                site, message = create_new_site(
+                    session=session,
+                    ml_id=len(site_uuid_list) + 1,
```
[the remark this answers is not in the document]
Comment (peterdudfield) [reply src/sites_toolbox.py:320]: https://github.com/openclimatefix/pv-site-api/blob/main/pv_site_api/main.py#L260

File: src/sites_toolbox.py [hunk 5]
```diff
@@ -308,22 +317,142 @@ def sites_toolbox_page():
...
-                    client_site_name=client_site_name,
-                    latitude=latitude,
-                    longitude=longitude,
-                    capacity_kw=capacity_kw,
-                    created_utc=created_utc,
-                )
-                st.write(message)
-                st.wrtie("New site details:", site)
+            max_ml_id = session.query(func.max(SiteSQL.ml_id)).scalar()
+            if max_ml_id is None:
```
Comment (peterdudfield) [inline src/sites_toolbox.py:321]: becasue this file is getting very long, is it worth breaking it up a bit like
- new directory called sites_toolbox
- file called create_site
- file called read_site_and_user
- file called update_site_group
or something like that? Maybe that can be in a seperate PR

Comment (rachel-labri-tipton) [reply src/sites_toolbox.py:321]: great idea.

**Review by peterdudfield — APPROVED** [review]

**Pull request merged**